Dockerfile

```diff
@@ -13,10 +13,16 @@ RUN apk add --no-cache \
         curl \
         openjdk17-jre-headless
 
-# Add files from the build
-ADD . /worker
-
-# and set the workign directory
+ENTRYPOINT ["./docker-entrypoint.sh"]
 WORKDIR /worker
 
-ENTRYPOINT ["./docker-entrypoint.sh"]
+# We want to ensure that we bust the docker build cache whenever the
+# contents of the repo change, i.e. there are new commits (to main).
+# Without busting the cache at this point then we would need to use
+# the --no-cache argument to docker build to ensure that it is refreshed
+# each time it is built - thereby losing the benefit of having the above
+# layers already in the cache.
+ADD https://github.com/BP3/c7-rest-worker/commits /dev/null
```

Comment:
I get what you're trying to do here, but this seem odd, and l suspect is not really the correct thing to do. Maybe this is a github action thing, but for a local build, you'll now rebuild when anyone pushes/merges even if you don't have those changes.

Reply:
This may be an imperfect implementation but I believe that the logic of cache busting is sound - either that or my book is out-of-date :-)

I did a bit more digging and found that 
> By default, GitHub Actions doesn't persist the Docker layer cache across builds
Thus, effectively, all builds use the `--no-cache` option at all times. So, in or out, this line won't actually make any difference to a pipeline. It will impact a local build in the way you say, but for someone who forgets to specify `--no-cache` that might actually be a good thing. Since my intention was to impact the pipeline then I will remove it.